ocaml/testsuite/tests/typing-layouts-float64/basics_alpha.ml

```diff
@@ -0,0 +1,904 @@
+[%%expect{|
+type t5_2_3 = { z : float; y : int; x : t_float64; }
+|}];;
+
+type t5_2_4 = { y : string; x : t_float64 };;
+[%%expect{|
+type t5_2_4 = { y : string; x : t_float64; }
+|}];;
+
+(* CR layouts 2.5: allow this *)
+type t5_3 = { x : t_float64 } [@@unboxed];;
+[%%expect{|
+Line 1, characters 14-27:
+1 | type t5_3 = { x : t_float64 } [@@unboxed];;
+                  ^^^^^^^^^^^^^
+Error: Type t_float64 has layout float64.
+       Unboxed records may not yet contain types of this layout.
+|}];;
+
+type t5_4 = A of t_float64;;
+[%%expect{|
+Line 1, characters 12-26:
+1 | type t5_4 = A of t_float64;;
+                ^^^^^^^^^^^^^^
+Error: Type t_float64 has layout float64.
+       Variants may not yet contain types of this layout.
+|}];;
+
+type t5_5 = A of int * t_float64;;
+[%%expect{|
+Line 1, characters 12-32:
+1 | type t5_5 = A of int * t_float64;;
+                ^^^^^^^^^^^^^^^^^^^^
+Error: Type t_float64 has layout float64.
+       Variants may not yet contain types of this layout.
+|}];;
+
+type t5_6 = A of t_float64 [@@unboxed];;
+[%%expect{|
+Line 1, characters 12-26:
+1 | type t5_6 = A of t_float64 [@@unboxed];;
+                ^^^^^^^^^^^^^^
+Error: Type t_float64 has layout float64.
+       Variants may not yet contain types of this layout.
+|}];;
+
+type ('a : float64) t5_7 = A of int
+type ('a : float64) t5_8 = A of 'a;;
+[%%expect{|
+type ('a : float64) t5_7 = A of int
+Line 2, characters 27-34:
+2 | type ('a : float64) t5_8 = A of 'a;;
+                               ^^^^^^^
+Error: Type 'a has layout float64.
+       Variants may not yet contain types of this layout.
+|}]
+
+type ('a : float64, 'b : float64) t5_9 = {x : 'a; y : 'b; z : 'a}
+
+type 'a t5_10 = 'a t_float64_id
+and 'a t5_11 = {x : 'a t5_10; y : 'a}
+
+type ('a : float64) t5_12 = {x : 'a; y : float#};;
+[%%expect{|
+type ('a : float64, 'b : float64) t5_9 = { x : 'a; y : 'b; z : 'a; }
+Line 4, characters 20-28:
+4 | and 'a t5_11 = {x : 'a t5_10; y : 'a}
+                        ^^^^^^^^
+Error: Layout mismatch in final type declaration consistency check.
+       This is most often caused by the fact that type inference is not
+       clever enough to propagate layouts through variables in different
+       declarations. It is also not clever enough to produce a good error
+       message, so we'll say this instead:
+         The layout of 'a is float64, because
+           of the definition of t_float64_id at line 2, characters 0-37.
+         But the layout of 'a must overlap with value, because
+           it instantiates an unannotated type parameter of t5_11, defaulted to layout value.
+       A good next step is to add a layout annotation on a parameter to
+       the declaration where this error is reported.
+|}];;
+
+type ('a : float64) t5_13 = {x : 'a; y : float#};;
+[%%expect{|
+type ('a : float64) t5_13 = { x : 'a; y : float#; }
+|}];;
+
+type 'a t5_14 = {x : 'a; y : float#};;
+[%%expect{|
+type 'a t5_14 = { x : 'a; y : float#; }
```

Comment:
I've left this test in `typing-layouts-float64`, and added the other suggested tests in `typing-layouts/mixed_records.ml` (which I moved from `mixed-blocks/mixed_records.ml` and added a few more tests to).

Reply:
I ended up moving more of these tests into `mixed_records.ml`, which now has a twin `mixed_records_alpha.ml`. I minted new files instead of reusing `basics.ml` because the output for some tests is different in runtime 4. (To sidestep this, `mixed_records*.ml` is currently just run for runtime 5.)